pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -93,9 +93,7 @@ func NewCmdRollingUpdate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().Duration("timeout", timeout, `Max time to wait for a replication controller to update before giving up. Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`)
 	usage := "Filename or URL to file to use to create the new replication controller."
 	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
-	cmd.MarkFlagRequired("filename")
```

Comment:
why remove this ?

Reply:
Because it is not a required flag. You can also issue the command with `kubectl rolling-update frontend-v1 -f xxx`.

Please refer to the template example in `rollingupdate.go`.